### PR TITLE
Add pre-commit config and run in CI

### LINKED ISSUE #127: Add license check pre-commit hook
See how https://github.com/ansys/grantami-serverapi-openapi/issues/173 is resolved and apply similar logic in this repo

### PR DESCRIPTION
**Pull request by Andy-Grigg**
Closes #127 

Add a pre-commit config to this repo. The changes here are all as a result of running `pre-commit run --all-files` before committing.

Also add `pre-commit run --all-files` to the code generation CI, replacing the current `black` invocation.

According to https://pre-commit.ci/, the CI service will fix PRs, but won't push for commits made by bots. I understand that to mean we need to invoke pre-commit manually in our CI.

### CODE REVIEW

File: .isort.cfg [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[settings]
```
Comment (Andy-Grigg) [inline .isort.cfg:1]: We could alternatively add this config and the black line-length to the code template. I'm not sure that makes sense though, ultimately the way the code is styled feels like something we might want to maintain separately.

File: .github/workflows/generate_library.yml [hunk 2]
```diff
@@ -59,11 +59,10 @@ jobs:
...
       working-directory: ansys-grantami-bomanalytics-openapi
 
-    - name: "Install and run black"
+    - name: "Install and run pre-commit"
       run: |
-        pip install black
-        black ./src ./tests
-      working-directory: ansys-grantami-bomanalytics-openapi
+        pip install pre-commit
+        ! pre-commit run --all-files
```
Comment (ludovicsteinbach) [inline .github/workflows/generate_library.yml:65]: It might be a stupid question, what does the "!" do?

**Review by ludovicsteinbach — APPROVED** [review]

**Review by da1910 — APPROVED** [review]
Looks reasonable to me. I wonder about pre-commit config to run `pre-process.ps1` but thats a question for another day

File: .github/workflows/generate_library.yml (hunk 2 above)
Comment (ludovicsteinbach) [reply .github/workflows/generate_library.yml:65]: ah I get it. I suppose this is ok, seeing as we know the hooks are never going to succeed

**Pull request merged**